bpf/bpf_overlay.c

```diff
@@ -190,9 +190,11 @@ int tail_handle_ipv6(struct __ctx_buff *ctx)
 #endif /* ENABLE_IPV6 */
 
 #ifdef ENABLE_IPV4
-static __always_inline int ipv4_host_delivery(struct __ctx_buff *ctx, struct iphdr *ip4)
+static __always_inline int ipv4_host_delivery(struct __ctx_buff *ctx,
+							struct iphdr *ip4,
+							int deliver_via_stack)
 {
-	if (1) {
+	if (!deliver_via_stack) {
```

Comment:
```suggestion
	if (deliver_via_stack) {
	        /* Deliver to the host-stack */
		ctx_change_type(ctx, PACKET_HOST);
		return CTX_ACT_OK;
	}
	
	if (1) {
	      ....
	}
```

Maybe it's just personal preference, but this reads easier to me.